addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/SOAPActionSpoofingActiveScanRule.java

```diff
@@ -73,91 +80,116 @@ public String getReference() {
         return Constant.messages.getString(MESSAGE_PREFIX + "refs");
     }
 
+    private TableWsdl getTable() {
+        return Control.getSingleton()
+                .getExtensionLoader()
+                .getExtension(ExtensionImportWSDL.class)
+                .getTable();
+    }
+
     @Override
     public int getCategory() {
         return Category.MISC;
     }
 
-    /*
-     * This method is called by the active scanner for each GET and POST parameter
-     * for every page
-     *
-     * @see
-     * org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.
-     * paros.network.HttpMessage, java.lang.String, java.lang.String)
+    /**
+     * This method is called by the active scanner for each GET and POST parameter for every page.
      */
     @Override
     public void scan() {
-        try {
-            /* Retrieves the original request-response pair. */
-            final HttpMessage originalMsg = getBaseMsg();
-            /* This scan is only applied to SOAP 1.1 messages. */
-            String currentHeader = originalMsg.getRequestHeader().getHeader("SOAPAction");
-            if (currentHeader != null && originalMsg.getRequestBody().length() > 0) {
-                currentHeader = currentHeader.trim();
-                /* Retrieves available actions to try attacks. */
-                String[] soapActions = ImportWSDL.getInstance().getSourceSoapActions(originalMsg);
+        /* Retrieves the original request-response pair. */
+        final HttpMessage originalMsg = getBaseMsg();
+        String originalSoapAction = SoapAction.extractFrom(originalMsg);
+        if (originalSoapAction == null) {
+            // Not a SOAP message
+            return;
+        }
 
-                boolean endScan = false;
-                if (soapActions == null || soapActions.length == 0) {
-                    // No actions to spoof
-                    LOG.info(
-                            "Skipping "
-                                    + getName()
-                                    + " because no actions were found. (URL: "
-                                    + originalMsg.getRequestHeader().getURI().toString()
-                                    + ")");
-                    return;
+        /* Retrieves available actions to try attacks. */
+        List<SoapAction> soapActions;
+        try {
+            soapActions = getTable().getSourceSoapActions(originalSoapAction);
+        } catch (DatabaseException e) {
+            LOG.warn("Could not retrieve SOAP actions from the database. Stopping scan.", e);
+            return;
+        }
+        if (soapActions == null || soapActions.isEmpty()) {
+            // No actions to spoof
+            LOG.info(
+                    "Skipping "
```

Comment:
I'd suggest using other word than "skip" (maybe ignore like above?), the active scan allows to skip scan rules, which means that no other message would be scanned (which is not the case here, just this one that's not going to be scanned).